service/auth.go

```diff
@@ -197,3 +209,24 @@ func (c *authClient) ValidateToken(ctx context.Context, requestToken string) (*J
 
 	return res, nil
 }
+
+// To check token still on db or no
+// if no, it means user already logout and token is not allowed to access
+func (s *AuthClient) IsTokenValid(ctx context.Context, params *entity.GetByTokensRequest) (isValid bool, err error) {
+
+	user, err := s.userRepository.GetByTokens(ctx, params)
+
+	if err != nil {
+		log.Printf("Error (IsTokenValid): %s", err.Error())
+		err = errTokenExpired
+		return
+	}
+
+	if user == nil {
```

Comment:
since if record not found is returning error, hence this line should be obsolete now